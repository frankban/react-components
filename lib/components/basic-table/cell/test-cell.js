/* Copyright (C) 2017 Canonical Ltd. */
'use strict';

const React = require('react');
const enzyme = require('enzyme');

const BasicTableCell = require('./cell');

describe('BasicTableCell', () => {

  const renderComponent = (options = {}) => enzyme.shallow(
    <BasicTableCell
      classes={options.classes}
      columnSize={options.columnSize || 5}
      content={options.content}
      isLastCol={options.isLastCol} />
  );

  it('can render', () => {
    const wrapper = renderComponent({
      content: (<span>Content!</span>)
    });
    expect(wrapper).toMatchSnapshot();
  });

  it('can render when last col', () => {
    const wrapper = renderComponent({
      isLastCol: true
    });
    assert.equal(wrapper.prop('className').includes('last-col'), true);
  });
});
