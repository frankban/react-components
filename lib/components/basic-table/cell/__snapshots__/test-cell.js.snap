// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BasicTableCell can render 1`] = `
<td
  className=""
>
  <span>
    Content!
  </span>
</td>
`;
